tools/remove_grouping_from_report_results.rb

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env ruby
+require 'trollop'
+
+opts = Trollop.options do
+  banner "Remove extras[:grouping] from MiqReportResult#report column.\n\nUsage: ruby #{$PROGRAM_NAME} [options]\n\nOptions:\n\t"
+  opt :dry_run,    "For testing, rollback any changes when the script exits.", :short => :none, :default => false
+  opt :batch_size, "Limit memory usage by process this number of report results at a time.",    :default => 50
+  opt :count,      "Stop checking after this number of report results.",                        :default => 0
+end
+
+puts "Using options: #{opts.inspect}\n\n"
+
+if defined?(Rails)
+  puts "Warning: Rails is already loaded!  Please do not invoke using rails runner. Exiting with help text.\n\n"
+  Trollop.educate
+end
+
+# Load rails after trollop options are set.  No one wants to wait for -h.
+require File.expand_path('../config/environment', __dir__)
+
+# Wrap all changes in a transaction and roll them back if dry-run or an error occurs.
+ActiveRecord::Base.connection.begin_transaction(:joinable => false)
+
+if opts[:dry_run]
+  puts "Running in dry-run, all changes will be rolled back when complete."
+
+  at_exit do
```

Comment:
Just because I am not sure:  I assume that `at_exit` will still fire after a `exit 1`, right?

(Nick could probably just go test this in a ruby script himself...)

Reply:
yeah, good point.  I'll test that

Reply:
No need, I already did:

```console
$ ruby -e "at_exit { puts 'foo' }; exit 42"
foo
$ echo $?
42
```

Reply:
I did several tests of dry run so I'm pretty confident it works.  It's a little messy how I'm doing it right now but I can't think of a better way to handle the begin/commit/rollback for dry-run/error/exit